### PR TITLE
AST-486 fix:Issue with social icon not change in customizer preview.

### PR DESCRIPTION
**Pull request by shubhw12**
### Description
Whenever we pick a new icon from the default list of icons the icon is not visible in the preview. It always shows the last saved icon.

### Screenshots

https://share.bsf.io/rRuGrdn0

### Types of changes
Bugfix (non-breaking change which fixes an issue)

### How has this been tested?

Checked the working of the social widget in the customizer.

### Checklist:
- [ ] My code is tested
- [ ] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -6,6 +6,8 @@ v3.6.9
 - Fix: WooCommerce - Cart icon broken on Tab & Mobile devices.
 - Fix: WooCommerce - Cart count is overlapping on submenu section.
 - Fix: Desktop logo width is applying on the mobile devices for a moment.
+- Fix: Social icon does not change in customizer preview.
+
```
Comment (imnavanath) [inline changelog.txt:10]:
```suggestion
```

File: inc/customizer/extend-custom-controls/src/social-icons/item-component.js [hunk 2]
```diff
@@ -20,6 +20,10 @@ const ItemComponent = props => {
 	const icon = props.item.id.replace(/[\d_]+$/g, ''); // Regex to replace numeric chars with empty string.
 	const urlLabel = ( 'phone' === props.item.id || 'phone_2' === props.item.id ) ? __('Number', 'astra') : __('URL', 'astra');
 
+	const[ selectedIcon , setselectedIcon ] = useState(
+		Icons[icon]
+	)
+	console.log(Icons[icon]);
```
Comment (imnavanath) [inline inc/customizer/extend-custom-controls/src/social-icons/item-component.js:26]:
```suggestion

```

File: inc/customizer/extend-custom-controls/src/social-icons/item-component.js [hunk 3]
```diff
@@ -20,6 +20,10 @@ const ItemComponent = props => {
 	const icon = props.item.id.replace(/[\d_]+$/g, ''); // Regex to replace numeric chars with empty string.
 	const urlLabel = ( 'phone' === props.item.id || 'phone_2' === props.item.id ) ? __('Number', 'astra') : __('URL', 'astra');
 
+	const[ selectedIcon , setselectedIcon ] = useState(
```
Comment (premanshup) [inline inc/customizer/extend-custom-controls/src/social-icons/item-component.js:23]:
```suggestion
	const[ selectedIcon , setSelectedIcon ] = useState(
```

File: changelog.txt (hunk 1 above)
Comment (shubhw12) [reply changelog.txt:10]: Done

File: inc/customizer/extend-custom-controls/src/social-icons/item-component.js (hunk 2 above)
Comment (shubhw12) [reply inc/customizer/extend-custom-controls/src/social-icons/item-component.js]: Resolved

File: inc/customizer/extend-custom-controls/src/social-icons/item-component.js (hunk 3 above)
Comment (shubhw12) [reply inc/customizer/extend-custom-controls/src/social-icons/item-component.js]: Done the specified changes

File: inc/customizer/extend-custom-controls/src/social-icons/item-component.js [hunk 4]
```diff
@@ -65,7 +73,10 @@ const ItemComponent = props => {
 				renderFunc= {renderSVG}
 				theme="default"
 				value={props.item.icon}
-				onChange={ value => { props.onChangeIcon(value, props.index); } }
+				onChange={ value => {
+					 props.onChangeIcon(value, props.index); 
+					 setSelectedIcon( Icons[value] );
+					} }
 				isMulti={false}
 				noSelectedPlaceholder= { __( "Select Icon" ) }
```
Comment (imnavanath) [inline inc/customizer/extend-custom-controls/src/social-icons/item-component.js:81]:
```suggestion
				noSelectedPlaceholder= { __( 'Select Icon', 'astra' ) }
```

**Comment by premanshup** [comment]
@shubhw12 Do check the GitHub action checks.

**Comment by premanshup** [comment]
Closing this PR. Code Analysis issue is resolved here - https://github.com/brainstormforce/astra/pull/3165

**Pull request closed without merging**